src/main/scala/beam/agentsim/agents/freight/input/PayloadPlansConverter.scala

```diff
@@ -21,76 +23,178 @@ import scala.util.Random
 /**
   * @author Dmitry Openkov
   */
-object PayloadPlansConverter {
-
-  def readFreightTours(path: String): Map[Id[FreightTour], FreightTour] = {
-    GenericCsvReader
-      .readAsSeq[FreightTour](path) { row =>
-        //tourId,departureTimeInSec,departureLocationX,departureLocationY,maxTourDurationInSec
-        val tourId: Id[FreightTour] = row.get("tourId").createId[FreightTour]
-        val departureTimeInSec = row.get("departureTimeInSec").toInt
-        val departureLocationX = row.get("departureLocationX").toDouble
-        val departureLocationY = row.get("departureLocationY").toDouble
-        val maxTourDurationInSec = row.get("maxTourDurationInSec").toInt
-        FreightTour(
-          tourId,
-          departureTimeInSec,
-          new Coord(departureLocationX, departureLocationY),
-          maxTourDurationInSec
-        )
+object PayloadPlansConverter extends LazyLogging {
+
+  val freightIdPrefix = "freight"
+
+  private def getRowValue(table: String, row: java.util.Map[String, String], key: String): String = {
+    if (row.containsKey(key)) {
+      row.get(key)
+    } else {
+      throw new IllegalArgumentException(s"Missing key '$key' in table '$table'.")
+    }
+  }
+
+  private def findClosestUTMPointOnMap(utmCoord: Coord, geoUtils: GeoUtils, streetLayer: StreetLayer): Option[Coord] = {
+    val wsgCoord = geoUtils.utm2Wgs(utmCoord)
+    val theSplit = geoUtils.getR5Split(streetLayer, wsgCoord)
+    if (theSplit == null) {
+      None
+    } else {
+      val wgsPointOnMap = geoUtils.splitToCoord(theSplit)
+      val utmCoord = geoUtils.wgs2Utm(wgsPointOnMap)
+      Some(utmCoord)
+    }
+  }
+
+  def readFreightTours(
+    freightConfig: BeamConfig.Beam.Agentsim.Agents.Freight,
+    geoUtils: GeoUtils,
+    streetLayer: StreetLayer
+  ): Map[Id[FreightTour], FreightTour] = {
+
+    val maybeTours = GenericCsvReader
+      .readAsSeq[Option[FreightTour]](freightConfig.toursFilePath) { row =>
+        def get(key: String): String = getRowValue(freightConfig.toursFilePath, row, key)
+        // tourId,departureTimeInSec,departureLocation_zone,maxTourDurationInSec,departureLocationX,departureLocationY
+        val tourId: Id[FreightTour] = get("tour_id").createId[FreightTour]
+        val departureTimeInSec = get("departureTimeInSec").toDouble.toInt
```

Comment:
Why not go directly to an `Int`?

Reply:
I think because using directly `toInt` cannot parse decimal values. I now added `round` to avoid truncating

Reply:
'departureTimeInSec' column contains doubles with fractional part equal to 0 (i.e. 1430.0)